### PR TITLE
Support length on Gray ranges

### PR DESCRIPTION
**Pull request by timholy**

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -825,4 +825,11 @@ ColorTypes.comp2(c::RGBA32) = alpha(c)
         @test copy(c) === c
     end
 
+    @testset "ranges" begin
+        T = Gray{N0f8}
+        @test length(zero(T):eps(T):oneunit(T)) == 256
+        @test length(zero(T):eps(N0f8):oneunit(T)) == 256
+        @test length(zero(N0f8):eps(T):oneunit(N0f8)) == 256
```
Comment (johnnychen94) [inline test/runtests.jl:832]: Maybe also add `@test step(r) = eps(T)`?

**Review by johnnychen94 — APPROVED** [review]
There might be more edge cases and range types if we want to fully support ranges. As an incremental patch, this one looks good to me. Or in other words, it is harmless to add these support.

**Comment by timholy** [comment]
We seem to be in good shape as far as subtypes defined in Base, but you're right that it's best to add the tests now to ensure it doesn't regress. The `StepRange` case is the hardest one because of the potential for overflow.
